src/screens/CreateGuild/index.tsx

```diff
@@ -0,0 +1,322 @@
+  const [selectedImage, setSelectedImage] = useState<Image>({} as Image);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const userSession: FirebaseAuthTypes.User = auth().currentUser!;
+
+  useEffect(() => {
+    let isMounted = true;
+
+    isMounted && isFocused && dispatch(setTitle('Create Guild'));
+
+    return () => {
+      isMounted = false;
+    };
+  }, [isFocused]);
+
+  const uploadBanner = async (): Promise<string | void> => {
+    const filename = selectedImage.filename;
+    const uri = selectedImage.uri;
+
+    try {
+      const imageRef = `${uuidv4()}.${getImageType(filename)}`;
+      const reference = storage().ref(imageRef);
+
+      await reference.putFile(uri);
+
+      return imageRef;
+    } catch (err: any) {
+      Alert.alert(
+        '>.<',
+        firebaseExceptions[err.code] || 'Não foi possível alterar o banner.',
+        [
+          {
+            text: 'Ok',
+          },
+        ],
+      );
+    }
+  };
+
+  const onSubmit = async (data: FormData) => {
+    setIsLoadingRequest(true);
+
+    try {
+      let imageRef: string | void;
+
+      if (Object.keys(selectedImage).length !== 0) {
+        imageRef = await uploadBanner();
+      }
+
+      const { id } = await firestore()
+        .collection('guilds')
+        .add({
+          name: data.name,
+          bannerRef: imageRef ? imageRef : '',
+          description: data.description,
+          members: [userSession.uid],
+          owner: userSession.uid,
+          createdAt: firestore.FieldValue.serverTimestamp(),
+          warCry: data.warCry,
+        });
+
+      await firestore()
+        .collection<ProfileDto>('profiles')
+        .doc(userSession.uid)
+        .update({
+          guild: id,
+        });
+
+      Alert.alert('=D', 'Guild criada com sucesso', [
+        {
+          text: 'Ok',
+          onPress: () => navigation.goBack(),
+        },
+      ]);
+    } catch (err) {
+      Alert.alert(
+        '>.<',
+        'Não foi possível concluir a solicitação, tente novamente mais tarde.',
+        [
+          {
+            text: 'Ok',
+          },
+        ],
+      );
+    } finally {
+      setIsLoadingRequest(false);
+    }
+  };
+
+  const handleGallery = async () => {
+    const permissionStatus = await getPermissions('gallery');
+
+    if (permissionStatus === 'granted') {
+      const imageFromStorage = await getPictureFromStorage();
+      const filename = imageFromStorage.assets?.[0].fileName;
+      const uri = imageFromStorage.assets?.[0].uri;
+
+      if (filename && uri) {
+        setSelectedImage({
+          filename,
+          uri,
+        });
+      }
+    }
+  };
+
+  const handleCamera = async () => {
+    const permissionStatus = await getPermissions('camera');
+
+    if (permissionStatus === 'granted') {
+      const imageFromCamera = await getPictureFromCamera();
+      const filename = imageFromCamera.assets?.[0].fileName;
+      const uri = imageFromCamera.assets?.[0].uri;
```

Comment:
same as last comment

Reply:
Done. I created a function to handle images from gallery e camera.